### PR TITLE
gut: init at 0.2.7

### LINKED ISSUE #224096: Package request: gut
**Project description**

Gut is a CLI designed to make Git easier to use.

**Metadata**

* homepage URL: [https://gut-cli.dev](https://gut-cli.dev)
* source URL: [https://github.com/julien040/gut](https://github.com/julien040/gut)
* license: mit
* platforms: linux, darwin, windows

### PR DESCRIPTION
**Pull request by paveloom**
###### Description of changes

https://gut-cli.dev

Closes #224096.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Rampoina** [comment]
Result of `nixpkgs-review pr 224267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gut</li>
  </ul>
</details>

**Comment by Rampoina** [comment]
Shouldn't this depend on git?

File: pkgs/applications/version-management/gut/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  doCheck = false;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "An easy-to-use git client for Windows, macOS, and Linux";
+    homepage = "https://github.com/slackhq/go-audit";
+    maintainers = [ lib.maintainers.paveloom ];
+    license = [ lib.licenses.mit ];
+    platforms = lib.platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/gut/default.nix:30]:
```suggestion
```
set by buildGoModule

**Comment by paveloom** [comment]
> Shouldn't this depend on git?

Well, for one, Git is required to use Nix flakes (I just learned that the hard way: by removing Git from my NixOS configuration).

Also, I don't see any other Git related package propagating the Git dependency (e.g., [`lazygit`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/lazygit/default.nix)).

**Pull request merged**